### DISCUSSION TITLE
Qual linguagem aprender depois do Javascript?

### DISCUSSION THREAD

**Discussion by thiamsantos** [frontendbr/forum#508] [Discussão]
Qual linguagem eu deveria aprender após estudar e já ter um bom nível de JavaScript?

Preferencialmente voltada para o Backend. O que vocês me recomendam, levando em consideração performance, sintaxe e paradigma utilizado pela linguagem?

**Comment by ericdouglas** [comment]
**Elixir**. 

Linguagem funcional, mas não tão "hardcore" como Haskell, feita em cima da EVM (Erlang Virtual Machine), projeto com 30 anos de bagagem e simplesmente _mind blowing_.

[Alguns](https://github.com/ericdouglas/dev-log/blob/master/source/elixir.md) [links](https://www.youtube.com/watch?v=h_NvYwveeMc&index=14&list=PLE7tQUdRKcyYoiEKWny0Jj72iu564bVFD) [interessantes](https://medium.com/@kenmazaika/why-im-betting-on-elixir-7c8f847b58#.m4q8db4xg).

[Essa talk](https://www.youtube.com/watch?v=h_NvYwveeMc&index=14&list=PLE7tQUdRKcyYoiEKWny0Jj72iu564bVFD) foi dada por [esse cara](http://boyd.multerer.com/about), ex-líder do time do Xbox, e nela ele faz uma projeção dos desafios que vamos ter nos próximos 10 anos na área de desenvolvimento de software em geral, e mostra como Elixir se sai muito bem nesses cenários, e é a tecnologia que ele está apostando.

**Comment by ghost** [comment]
Estou cada vez acreditando mais em Elixir por ter uma VM bem madura e Go por ter o Google por trás. Atualmente estou estudando Elixir como nova linguagem acho que tem muito a crescer. Poderia dizer que a linguagem é fácil desde que você manje dos conceitos funcionais. Então fica ela como dica também.

Um ponto de atenção é que Elixir não tem tanta coisa pronta como é em outras linguagens (javascript, ruby, c#, java) mas dá pra sobreviver.

Globo, Locaweb e outras empresas no Brasil vem apostando em Elixir, fica aqui também um link interessante:

http://hipsters.tech/por-tras-dos-videos-da-globo-com-hipsters-11/

O Emerson Macedo da Globo esteve na Ruby Conf 2016 falando sobre como eles estão usando Elixir na plataforma de videos deles. É fantástico os números que eles conseguiram fazendo apenas uma "rota" da aplicação rsrs.

**Comment by woliveiras** [comment]
@thiamsantos, se for investir um tempo em Elixir, aqui tem um excelente ponto de aprendizado: http://elixirschool.com/

**Comment by barbier** [comment]
Vou contra a maré e recomendar PHP, Java ou C#.
PHP que evoluiu bastante na versão 7 e possui frameworks com comunidade bastante ativa.
Java que também ficou bem interessante, pro desenvolvedor, na última versão, a 8.
C# que agora é parcialmente open-source, possui uma comunidade de desenvolvedores relativamente ativa e ganhou diversos usos.

**Comment by woliveiras** [comment]
Pra arrumar um trampo como Back Endero ou FullStackerzudo é mais fácil com as linguagens que o @barbier comentou mesmo, viu.

Nesse caso eu iria de Java pra ganhar mais dinheiros!!! 🤑

**Comment by leonardoelias** [comment]
@woliveiras não acho que Java ganhe mais dinheiro, até porque ultimamente está bem mais difícil achar  vaga para Java, ou estou enganado ?

**Comment by kelciocajueiro** [comment]
@leonardoelias recebo quase todo dia e-mails de vagas Java aqui em Brasília.

**Comment by woliveiras** [comment]
@leonardoelias, rapaz, aqui em SP vejo muita vaga de Java e os Javeiros tem ganhado bem viu.

**Comment by adrianoavelino** [comment]
Essa pesquisa da trampos com a umbler pode ajudar?

http://tutano.trampos.co/14187-infografico-mercado-de-desenvolvimento-brasil/

**Comment by mtmr0x** [comment]
Cara, a que tu quiseres. (Eu ouvi amém?)

Se tua vontade é pro mercado, dá uma olhada no link do @adrianoavelino ali.

Se tua vontade é simplesmente aprender alguma coisa, escolha uma que atende o seu maior sonho como projeto.

Beijos de luz

**Comment by JoseCage** [comment]
Tem um outro infográfico (da **umbler** + **trampos.co**). Dá uma olhada [aqui](http://tutano.trampos.co/8228-infografico-qual-linguagem-de-programacao-devo-aprender-primeiro/)

**Comment by Cleysonlb** [comment]
@mtmr0x concordo com você!! se for pro mercado faz uma pesquisa, se for pra você escolha oq vai te dar mais prazer aprendendo!

**Comment by thiamsantos** [comment]
Muitos obrigado a todos pelas dicas!!!
Eu já estudei o básico de [Clojure](https://clojure.org/) e [Haskell](https://www.haskell.org/), mas a sintaxe estranhava bastante. Com o Elixir parece ser diferente :heart:.
Vou procurar estudá-lo.

**Comment by didiraja** [comment]
@adrianoavelino muito legal o infográfico, obrigado

**Comment by matheusml** [comment]
Para o Back-End: **Elixir**
Para o Front-End: **Elm**
